### DISCUSSION TITLE
Positioning of text on Buttons.

### DISCUSSION THREAD

**Discussion by ColinHMurdoch** [Bodmer/TFT_eSPI#1636] [Q&A - General] [answered]
Hi Bodmer,

I have now written a project using ESP32 and a 3.5" TFT/LCD screen.  Imhave three screens with various buttons,  On all the buttons the text appears towards to top of the button - I would say about 2/3 of the height above the button.  Are there any settings which would align the text at the centre of the button?

Also I changed the Button Class to allow longer button names, but I see there is a reference to a feature called "long-names".  What is the correct usage for this please?

Regards,
Colin Murdoch.

**Comment by Bodmer** [comment, marked as answer]
Have a look at this example:
https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Generic/TFT_Button_Label_Datum

The datum parameter is the same as the setTextDatum function.

**Reply by ColinHMurdoch** [reply]
Many thanks Bodmer.  I knew you'd have an answer. I don’t think I have noticed that example before, so I will certainly try it out.  It looks like it will answer my questions.
Regards,
Colin.

**Reply by ColinHMurdoch** [reply]
Many thanks Bodmer.  That example sorted me out.
